src/node/historical_queries_adapter.cpp

```diff
@@ -374,4 +425,132 @@ namespace ccf::historical
       ccf::endpoints::EndpointFunction,
       ccf::endpoints::EndpointContext>(f, node_context, available, extractor);
   }
+
+  template <
+    class TQueryHandler,
+    class TEndpointFunction,
+    class TEndpointContext>
+  TEndpointFunction _adapter_v4(
```

Comment:
To avoid repetition, can we implement `_adapter_v3` in terms of `_adapter_v4`? I think that's just:

```
template <...>
TEndpointFunction _adapter_v3(...)
{
  return _adapter_v4(..., default_error_handler);
}
```